pkg/sql/sqlbase/structured.go

```diff
@@ -3272,13 +3272,46 @@ func (desc *MutableTableDescriptor) performComputedColumnSwap(swap *ComputedColu
 	// oldCol still needs to have values written to it in case nodes read it from
 	// it with a TableDescriptor version from before the swap.
 	// To achieve this, we make oldCol a computed column of newCol.
-	oldColComputeExpr := tree.CastExpr{
-		Expr:       &tree.ColumnItem{ColumnName: tree.Name(oldCol.Name)},
-		Type:       oldCol.DatumType(),
-		SyntaxMode: tree.CastShort,
+	// If a USING EXPRESSION was provided, we cannot generally invert
+	// the EXPRESSION to insert into the old column and thus will force an
+	// error using the computed expression. Any inserts into the new column
+	// will fail until the old column is dropped.
+	if swap.InverseExpr != nil && *swap.InverseExpr == tree.Serialize(tree.DNull) {
+		errMsg := fmt.Sprintf(
+			"Column %s is undergoing the ALTER COLUMN TYPE USING EXPRESSION "+
+				"schema change. Inserts are not supported until the schema change is "+
+				"finalized", oldCol.Name)
+		insertedValToString := tree.CastExpr{
+			Expr:       &tree.ColumnItem{ColumnName: tree.Name(oldCol.Name)},
+			Type:       types.String,
+			SyntaxMode: tree.CastShort,
+		}
+		insertedVal := tree.Serialize(&insertedValToString)
+		// Set the computed expression to use crdb_internal.force_error() to
+		// prevent writes into the column. Whenever the new column is written to
+		// crdb_internal.force_error() will cause the write to error out.
+		// This prevents the column going ALTER COLUMN TYPE to be unable to write
+		// to until the old column is dropped.
+		// This is safe to do so because after the column swap, the old column
+		// should become "read-only", new inserts should only be possible via
+		// the computed expression.
+		// An example of the error:
```

Comment:
I meant a string of the computed expression -- the error message already printed above.